### DISCUSSION TITLE
Getting data from LYWSD03MMC sensor with BTHome protocol on NimBLE ESP32

### DISCUSSION THREAD

**Discussion by zejulio** [h2zero/NimBLE-Arduino#664] [Q&A] [answered]
Hello,

I'm new to BLE and want to get the data from BTHome sensors, like LYWSD03MMC with ATC firmware on a ESP32.
I did not saw an example regarding this, and I am struggling to make it work...
All I want is to use the device MAC to listem to his broadcast messages, so it uses the less battery of the device.
The ESP32 catches that broadcasts and display on local display.
Can anyone point in the right direction?

Maybe with a small example, I already have the display part figure out, just need the "listen" part of NimBLE.

Thanks!

**Comment by zejulio** [comment]
I think I've managed to make it work this way.
Add my device to whitelist, do a search, and decode the advertise data.
This is workarround, since I think the device sends data periodically, and I wanted to put the ESP listening and get this data without having do to a search...
Am I wrong?
Thanks

**Comment by h2zero** [comment, marked as answer]
I think all you need to do is follow the continuous scan example and check the mac address in the callback.

**Comment by chinswain** [comment]
There's a library for this: https://github.com/matthias-bs/ATC_MiThermometer
